src/datadoc/backend/utils.py

```diff
@@ -191,62 +194,74 @@ def incorrect_date_order(
     return date_from is not None and date_until is not None and date_until < date_from
 
 
-def _has_metadata_value(
-    metadata: model.Dataset | model.Variable,
+def _is_missing_multilanguage_value(
+    field_name: str,
+    field_value,  # noqa: ANN001 Skip type hint to enable dynamically handling value for LanguageStringType not indexable
     obligatory_list: list,
-) -> list:
-    """Return metadata fields with value.
-
-    Args:
-        metadata: The metadata object to check.
-        obligatory_list: A list of obligatory fields.
-
-    Returns:
-        List of metadata fields that have values.
-    """
-    return [
-        k
-        for k, v in metadata.model_dump().items()
-        if k in obligatory_list and v is not None
-    ]
-
+) -> bool:
+    """Check obligatory fields with multilanguage value.
 
-def num_obligatory_dataset_fields_completed(dataset: model.Dataset) -> int:
-    """Count the number of obligatory dataset fields completed for each variable.
-
-    This function returns the total count of obligatory fields in the dataset that
-    have values (are not None).
+    This function checks whether a given field, which is supposed to have
+    multilanguage values, is missing values in all specified languages.
 
     Args:
-        dataset: The dataset object for which to count the fields.
+        field_name: The name of the field to check.
+        field_value: The value of the field. Expected to be of type LanguageStringType.
+        obligatory_list: A list of obligatory field names that should have multilanguage values.
 
     Returns:
-        The number of obligatory dataset fields that have been completed (not None).
+        True if no value in any of languages for one field, False otherwise.
     """
-    return len(
-        _has_metadata_value(dataset, OBLIGATORY_DATASET_METADATA_IDENTIFIERS),
-    )
+    if (
+        field_name in obligatory_list
+        and field_value
+        and (
+            len(field_value[0]) > 0
+            and not field_value[0]["languageText"]
+            and (len(field_value) <= 1 or not field_value[1]["languageText"])
+            and (
+                len(field_value) <= 2  # noqa: PLR2004 approve magic value
+                or not field_value[2]["languageText"]
+            )
+        )
+    ):
+        return True
+    return False
 
 
-def num_obligatory_variables_fields_completed(variables: list) -> int:
-    """Count the number of obligatory fields completed for each variable.
+def _has_multilanguage_value(
```

Comment:
Is this not the logical inverse of `_is_missing_multilanguage_value`? Do we need both of these functions?